cmd/limits.go

```diff
@@ -0,0 +1,117 @@
+	memoryArg string
+	memory    k8sResource.Quantity
+}
+
+var limitConfig limitFlags
+
+func (limit *limitFlags) parseLimitFlags() {
+	if len(limit.cpuArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(limit.cpuArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		limit.cpu = quantity
+	}
+
+	if len(limit.memoryArg) != 0 {
+		quantity, err := k8sResource.ParseQuantity(limit.memoryArg)
+		if err != nil {
+			log.Error("Wrong cpu argument: " + err.Error())
+			return
+		}
+		limit.memory = quantity
+	}
+}
+
+func checkLimits(container Container, limits limitFlags, result *Result) {
+	if container.Resources.Limits == nil {
+		occ := Occurrence{id: ErrorResourcesLimitsNIL, kind: Warn, message: "Resource limit not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	checkCPULimit(container, limits, result)
+	checkMemoryLimit(container, limits, result)
+}
+
+func checkCPULimit(container Container, limits limitFlags, result *Result) {
+	cpu := container.Resources.Limits.Cpu()
+	if cpu == nil || cpu.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsCpuNIL, kind: Warn, message: "CPU limit not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if limits.cpu.MilliValue() > 0 && cpu.MilliValue() > limits.cpu.MilliValue() {
+		message := fmt.Sprintf("CPU limit exceeded, it is set to %s but it must not exceed %s. Please adjust it!", cpu.String(), limits.cpu.String())
+		occ := Occurrence{id: ErrorResourcesLimitsCpuExceeded, kind: Warn, message: message}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+}
+
+func checkMemoryLimit(container Container, limits limitFlags, result *Result) {
+	memory := container.Resources.Limits.Memory()
+	if memory == nil || memory.IsZero() {
+		occ := Occurrence{id: ErrorResourcesLimitsMemoryNIL, kind: Warn, message: "Memory limit not set, please set it!"}
+		result.Occurrences = append(result.Occurrences, occ)
+		return
+	}
+
+	if limits.memory.Value() > 0 && memory.Value() > limits.memory.Value() {
+		message := fmt.Sprintf("Memory limit exceeded, it is set to %s but it must not exceed %s. Please adjust it!", memory.String(), limits.memory.String())
+		occ := Occurrence{id: ErrorResourcesLimitsMemoryExceeded, kind: Warn, message: message}
+		result.Occurrences = append(result.Occurrences, occ)
+	}
+}
+
+func auditLimits(limits limitFlags, items Items) (results []Result) {
+	limits.parseLimitFlags()
+
+	for _, item := range items.Iter() {
+		containers, result := containerIter(item)
+		for _, container := range containers {
+			checkLimits(container, limits, result)
+			if result != nil && len(result.Occurrences) > 0 {
+				results = append(results, *result)
+				break
+			}
+		}
+	}
+	return
+}
+
+var limitsCmd = &cobra.Command{
+	Use:   "limits",
+	Short: "Audit containers running with limits",
+	Long: `This command determines which containers in a kubernetes cluster
```

Comment:
👋 Hey thanks for putting up an awesome PR, just wanted to add that the description should probably be something closer to `This command determines which containers in a kubernetes cluster have and do not exceed specified cpu and memory limits`. The command does more than just verify the limits' existence.

Reply:
Thank you, I didn't update the description after adding the parameters.